compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -179,6 +200,19 @@ impl<'a> Parser<'a> {
         &self.sess.span_diagnostic
     }
 
+    pub(super) fn restore(&mut self, snapshot: SnapshotParser<'a>) {
+        *self = snapshot.parser;
+        self.unclosed_delims.extend(snapshot.unclosed_delims.clone());
+    }
+
+    pub(super) fn diagnostic_snapshot(&self) -> SnapshotParser<'a> {
```

Comment:
Likewise, a documentation comment on this function would be great.

```suggestion
    pub(super) fn create_snapshot_for_diagnostics(&self) -> SnapshotParser<'a> {
```